### ISSUE TITLE
local associative array seen as global

### ISSUE THREAD

**Issue by wybodekker** [koalaman/shellcheck#2171]
#### For bugs
- Rule Id: SC2190
- My shellcheck version (`shellcheck --version` or 'online'): online and local version 0.7.1
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

#!/bin/bash

f() {
   local -A map
   map[x]=assoc
   map[y]=array
   echo "local map: ${map[*]}"
}

g() {
   local -a map
   map+=(index)
   map+=(arr)
}

f
echo "x: ${map[x]} y: ${map[y]}"
g
echo "global map: ${map[*]}"
```

#### Here's what shellcheck currently says:
```
Line 12	SC2190: Elements in associative arrays need index, e.g. array=( [index]=value ) .
Line 13	SC2190: Elements in associative arrays need index, e.g. array=( [index]=value ) .
```
#### Here's what I wanted or expected to see:
it should give error code SC2034 instead: map appears unused. Verify...

**Comment by pmhahn** [comment]
```sh
# shellcheck shell=bash
foo () {
        declare -a cmd=(1 2 3)
        echo "func ${cmd[*]}"
}
cmd="bar"
foo
echo "main ${cmd[*]}"
```
results in
> Line 6:
> cmd="bar"
> ^-- SC2178: Variable was used as an array but is now assigned a string.

while `cmd` is local to the function `foo` only and does not spill out
